inspire/base/static/js/search/search_results.js

```diff
@@ -44,8 +44,176 @@ require(['jquery', 'bootstrap'], function($) {
         $("#download" + data.recid).attr("href", "/formatter/download-latex/latex_us/" + data.recid)
       });
     });
+
   })
 
+  $('#formats-dropdown-menu > li').each(function(index) {
+    $(this).on("click", function() {
+      $("#dropdown-export").html($(this).text() + ' <span class="caret"></span>');
+    });
+  });
+
+  var sList = [];
+  $('#checkbox-select-all').click(function(event) {
+    if (this.checked) {
+      $('.checkbox-results').each(function() {
+        this.checked = true;
+        sList.push($(this).attr("id"));
+      });
+      $('#results-panel').after('<div class="panel panel-default" id="info-message">' +
+        '<div class="panel-body" >You have selected ' + filter_list(sList).length + ' records of this page. <a class="pointer" id="select-all-records">' +
+        'Select all ' + $('#total-results').text() + ' results.</a></div></div>');
+      $('#select-all-records').on('click', function() {
+        $('#info-message').remove();
+        $('#results-panel').after('<div class="alert alert-warning" id="alert-selection" role="alert">' +
+          $('#total-results').text() + ' have been selected.<a class="pointer" id="undo-selection"> Undo selection.</a></div>');
+        $.get("/search?of=id&rg=" + $('#total-results').text(), function(data, status) {
```

Comment:
Thats the idea, but I didn't implement the limit. I will add it now. So if
there are more than 5000 records we cut it at 5000. Correct?

2015-10-30 11:17 GMT+01:00 Javier Martin Montull <notifications@github.com>:

> In inspire/base/static/js/search/search_results.js
> <https://github.com/inspirehep/inspire-next/pull/450#discussion_r43487708>
> :
>
> > +
> > +  var sList = [];
> > +  $('#checkbox-select-all').click(function(event) {
> > +    if (this.checked) {
> > +      $('.checkbox-results').each(function() {
> > +        this.checked = true;
> > +        sList.push($(this).attr("id"));
> > +      });
> > +      $('#results-panel').after('<div class="panel panel-default" id="info-message">' +
> > +        '<div class="panel-body" >You have selected ' + filter_list(sList).length + ' records of this page. <a class="pointer" id="select-all-records">' +
> > +        'Select all ' + $('#total-results').text() + ' results.</a></div></div>');
> > +      $('#select-all-records').on('click', function() {
> > +        $('#info-message').remove();
> > +        $('#results-panel').after('<div class="alert alert-warning" id="alert-selection" role="alert">' +
> > +          $('#total-results').text() + ' have been selected.<a class="pointer" id="undo-selection"> Undo selection.</a></div>');
> > +        $.get("/search?of=id&rg=" + $('#total-results').text(), function(data, status) {
>
> So $.get("/search?of=id&rg=" sometimes does not return succcess? And that
> means that there are too many records to download?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inspirehep/inspire-next/pull/450/files#r43487708>.
>